### ISSUE TITLE
Add URL whitelist/blacklist and URL request/response API

### ISSUE THREAD

**Issue by adamjs** [ultralight-ux/ultralight#84]
We should allow users to restrict URL requests and also do custom-handling for resource requests.

**Comment by KenanSulayman** [comment]
https://github.com/ultralight-ux/ultralight/issues/78

**Comment by APerricone** [comment]
is It about create something like [WebViewClient.shouldInterceptRequest](https://developer.android.com/reference/android/webkit/WebViewClient#shouldInterceptRequest(android.webkit.WebView,%20android.webkit.WebResourceRequest)) ?

**Comment by adamjs** [comment]
Yes-- the goal is to provide functionality similar to Chromium's `shouldInterceptRequest` so that users can monitor / modify / block requests as they are sent and optionally "intercept" them and respond with their own custom loader. (This supercedes the FileSystem API and allows custom handling of all requests)

I've already scoped out an implementation and API for this. Going to try to get this in for the 1.3 milestone.
